ctpolicy/chromepolicy.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ctpolicy
+
+import (
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// googGroupBuilder is used to build log-group covering Google-operated logs.
+type googGroupBuilder struct {
+}
+
+func (g googGroupBuilder) Name() string {
+	return "Google-operated"
+}
+func (g googGroupBuilder) IsMember(log *loglist.Log) bool {
+	return log.GoogleOperated()
+}
+func (g googGroupBuilder) IsBase() bool {
+	return false
+}
+
+// nonGoogGroupBuilder is used to build log-group covering not-Google-operated logs.
+type nonGoogGroupBuilder struct {
+}
+
+func (g nonGoogGroupBuilder) Name() string {
+	return "Non-Google-operated"
+}
+func (g nonGoogGroupBuilder) IsMember(log *loglist.Log) bool {
+	return !log.GoogleOperated()
+}
+func (g nonGoogGroupBuilder) IsBase() bool {
+	return false
+}
+
+// ChromeCTPolicy provides policy logic in appliance to the data provided.
+type ChromeCTPolicy struct {
+	googBuilder  googGroupBuilder
+	nGoogBuilder nonGoogGroupBuilder
+	baseBuilder  baseGroupBuilder
```

Comment:
None of these builders have state, so it feels slightly odd to hold onto instances of them.

Reply:
They contain map between names and group-logic. E.g. All-logs group is shared across 2policies.

Reply:
I don't follow -- I see three empty `struct`s:
```golang
type googGroupBuilder struct {
}
type nonGoogGroupBuilder struct {
}
type baseGroupBuilder struct {
}
```
Where is the map?